adapter: Update RBAC defaults

jkosh44 opened the pull request:
This PR updates the default value of the RBAC system variables to true.

### Motivation
This PR adds a feature that has not yet been specified.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  <!-- Reference the design in the description. -->
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  <!-- Ask in #team-cloud on Slack if you need help preparing the cloud PR. -->
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Enable RBAC for all new environments.

jkosh44 commented:
I accidentally screwed something up with the reviewers. Nikhil, if you see this then I added you by accident and you can ignore.

ParkMyCar commented on src/stash/src/upgrade/v35_to_v36.rs:
```diff
@@ -0,0 +1,196 @@
+                let mut found_user_flag = false;
+
+                for (key, value) in entries {
+                    if key.name == ENABLE_LD_RBAC_CHECKS_NAME {
+                        found_ld_flag = true;
```
Just a heads up I think this is true for all environments, since we're the [default rule for the flag](https://app.launchdarkly.com/default/production/features/enable_ld_rbac_checks/targeting) is `true`

ParkMyCar reviewed (approved):
Woooo!! Releasing RBAC! 🎉

jkosh44 commented:
I removed the `{{< private-preview />}}` tag from all the RBAC docs, hopefully that's not too controversial. We're going to wait until Thursday before the release to merge this, to see if there's any additional feedback from customers using RBAC.

ParkMyCar commented:
> I removed the `{{< private-preview />}}` tag from all the RBAC docs, hopefully that's not too controversial. We're going to wait until Thursday before the release to merge this, to see if there's any additional feedback from customers using RBAC.

Do we want to replace them with `{{< public-preview />}}`, or is RBAC considered GA?

jkosh44 commented:
> I removed the `{{< private-preview />}}` tag from all the RBAC docs, hopefully that's not too controversial. We're going to wait until Thursday before the release to merge this, to see if there's any additional feedback from customers using RBAC.

> > I removed the `{{< private-preview />}}` tag from all the RBAC docs, hopefully that's not too controversial. We're going to wait until Thursday before the release to merge this, to see if there's any additional feedback from customers using RBAC.
> 
> Do we want to replace them with `{{< public-preview />}}`, or is RBAC considered GA?

We discussed this in person, but just for the record, it should have technically already been in public preview, now that it will be available to everyone by default it is technically GA.

jkosh44 commented:
@ParkMyCar The force push was fixing some cargo tests that were failing because they had assumed RBAC would be off. I initially misread the failures and thought it was the Dynamic ID allocator test flake.
